homeassistant/components/amcrest/__init__.py

```diff
@@ -94,20 +100,28 @@ def setup(hass, config):
     """Set up the Amcrest IP Camera component."""
     from amcrest import AmcrestCamera, AmcrestError
 
-    hass.data[DATA_AMCREST] = {}
+    hass.data.setdefault(DATA_AMCREST, {})
     amcrest_cams = config[DOMAIN]
 
     for device in amcrest_cams:
+        name = device[CONF_NAME]
+        if name in hass.data[DATA_AMCREST]:
```

Comment:
Add a custom validator that checks that all names are unique. See the mysensors component for example of custom validator.

Reply:
Will do.